solidity/contracts/wrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -0,0 +1,165 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+import "./openzeppelin/Address.sol";
+import "./interfaces/IWrbtcERC20.sol";
+import "./interfaces/ILiquidityPoolV2Converter.sol";
+import "./interfaces/ISmartToken.sol";
+import "./interfaces/ISovrynSwapNetwork.sol";
+import "./interfaces/IERC20Token.sol";
+
+contract RBTCWrapperProxy{
+    
+    using Address for address;
+
+    IWrbtcERC20 wrbtc;
+    ILiquidityPoolV2Converter liquidityPoolV2Converter;
+    ISmartToken poolToken;
+    ISovrynSwapNetwork sovrynSwapNetwork;
+    IERC20Token docToken;
+    
+    address public wrbtcTokenAddress;
+    address public liquidityPoolV2ConverterAddress;
+    address public poolTokenAddress;
+    address public sovrynSwapNetworkAddress;
+    address public docTokenAddress;
+    
+    /**
+     * @dev To check if ddress is contract address 
+     */
+    modifier checkAddress(address address_) {
+        require(address_.isContract(), "The address is not a contract");
+        _;
+    }
+    
+    constructor (
+        address _wrbtcTokenAddress, 
+        address _liquidityPoolV2ConverterAddress, 
+        address _poolTokenAddress,
+        address _sovrynSwapNetworkAddress,
+        address _docTokenAddress
+        ) 
+        public 
+        checkAddress(_wrbtcTokenAddress)
+        checkAddress(_liquidityPoolV2ConverterAddress)
+        checkAddress(_poolTokenAddress)
+        checkAddress(_sovrynSwapNetworkAddress)
+        checkAddress(_docTokenAddress)
+        {
+            wrbtcTokenAddress = _wrbtcTokenAddress;
+            liquidityPoolV2ConverterAddress = _liquidityPoolV2ConverterAddress;
+            poolTokenAddress = _poolTokenAddress;
+            sovrynSwapNetworkAddress = _sovrynSwapNetworkAddress;
+            docTokenAddress = _docTokenAddress;
+
+            wrbtc = IWrbtcERC20(wrbtcTokenAddress);
+            liquidityPoolV2Converter = ILiquidityPoolV2Converter(liquidityPoolV2ConverterAddress);
+            poolToken = ISmartToken(poolTokenAddress);
+            sovrynSwapNetwork = ISovrynSwapNetwork(sovrynSwapNetworkAddress);
+            docToken = IERC20Token(docTokenAddress);
+    }
+
+    function() external payable { }
+
+    /**
+     * @dev  
+     * The process:
+     * 1.Accepts RBTC
+     * 2.Sends RBTC to WRBTC contract in order to wrap RBTC to WRBTC
+     * 3.Calls 'addLiquidity' on LiquidityPoolConverterV2 contract
+     * 4.Transfers pool tokens to user
+     * 
+     * @param _amount       amount of liquidity to add
+     * @param _minReturn    minimum return-amount of reserve tokens
+     *
+     * @return amount of pool tokens minted
+     */
+    function addLiquidity(uint256 _amount, uint256 _minReturn) payable public returns(uint256) {
```

Comment:
it would be nicer to let the user pass the converter and pool token address instead of reading it from storage. this way, we can use the same wrapper contract for all future converters (which we're going to have soon).